### PR TITLE
fix the way legacycsprojpackageref projects get target frameworks

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fixes: https://github.com/NuGet/Home/issues/4806

CC: @jainaashish @emgarten @alpaix @mishra14 @nkolev92 @rrelyea

### CODE REVIEW

File: src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs [hunk 1]
```diff
@@ -190,9 +192,41 @@ public NuGetFramework TargetNuGetFramework
         {
             get
             {
-                ThreadHelper.ThrowIfNotOnUIThread();
+                if (_nugetFramework != null)
```
Comment (emgarten) [inline src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs]: A more common pattern for the same thing would be:

```
if (x == null) {
   x = value;
}

return x;
```

File: src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs [hunk 2]
```diff
@@ -190,9 +192,41 @@ public NuGetFramework TargetNuGetFramework
...
+                {
+                    return _nugetFramework;
```
Comment (emgarten) [inline src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs]: Will re-targeting between RS2 and RS3 in the future reload the project? Is this safe to cache?

**Review by emgarten — CHANGES_REQUESTED** [review]
Add an E2E test for this

**Review by jainaashish — APPROVED** [review]
Looks good. Going forward we must add tests for legacy csproj uwp projects.

**Comment by rrelyea** [comment]
Issue: https://github.com/NuGet/Home/issues/4806

File: src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs (hunk 2 above)
Comment (rohit21agrawal) [reply src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs]: yes reloading will always happen on re-targeting.

**Review by emgarten — APPROVED** [review]
:rocket:

**Comment by rohit21agrawal** [comment]
checked into release-4.1.0-rtm via https://github.com/NuGet/NuGet.Client/commit/7d1e1e81d2e6aff47afaf6965659c597ce713a9f

**Comment by rohit21agrawal** [comment]
writing the test as a separate PR.

**Pull request merged**